tests/test_partition.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+from streaming.base.partition import get_partitions
+
+
+def test_partition_walk():
+    partition_algo = 'orig'
+    num_samples = 1000
+    num_canonical_nodes = 176
+    num_physical_nodes = 22
+    ranks_per_node = 8
+    workers_per_rank = 8
+    batch_size = 10
+    drop_first = 0
+    """
+    # Requires a lot of RAM and time
+    partition_algo = 'orig'
+    num_samples = 1999990222
+    num_canonical_nodes = 176
+    num_physical_nodes = 22
+    ranks_per_node = 8
+    workers_per_rank = 8
+    batch_size = 10
+    drop_first = 408112000
+    """
+
+    for drop_first in range(0, 500):
+        x = get_partitions(partition_algo, num_samples, num_canonical_nodes, num_physical_nodes,
+                           ranks_per_node, workers_per_rank, batch_size, drop_first)
+        assert x.shape == (22, 8, 8, 1, 10)
```

Comment:
Thanks! One more request, can you also validate the number of samples (after drop_first) to validate that we are not dropping unnecessary samples ? I think based on the above shape `(22, 8, 8, 1, 10)`, most of the samples are the placeholder (-1). So validate the actual number of samples after drop_first would be good testing too!